pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -647,51 +647,78 @@ func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.Applicat
+			if err != nil {
+				logrus.Debugf("error in reading %v cm, defaulting restore volume batch count", k8sutils.StorkControllerConfigMapName)
+			} else {
+				if len(restoreVolumeBatchCount) != 0 {
+					batchCount, err = strconv.Atoi(restoreVolumeBatchCount)
+					if err != nil {
+						logrus.Debugf("error in conversion of restoreVolumeBatchCount: %v", err)
+					}
+				}
+			}
+
+			for i := 0; i < len(backupVolInfos); i += batchCount {
+				batchVolInfo := backupVolInfos[i:min(i+batchCount, len(backupVolInfos))]
+				restoreVolumeInfos, err := driver.StartRestore(restore, batchVolInfo, preRestoreObjects)
+				if err != nil {
+					message := fmt.Sprintf("Error starting Application Restore for volumes: %v", err)
+					log.ApplicationRestoreLog(restore).Errorf(message)
+					if _, ok := err.(*volume.ErrStorageProviderBusy); ok {
+						msg := fmt.Sprintf("Volume restores are in progress. Restores are failing for some volumes"+
+							" since the storage provider is busy. Restore will be retried. Error: %v", err)
+						a.recorder.Event(restore,
+							v1.EventTypeWarning,
+							string(storkapi.ApplicationRestoreStatusInProgress),
+							msg)
+
+						log.ApplicationRestoreLog(restore).Errorf(msg)
+						// Update the restore status even for failed restores when storage is busy
+						_, updateErr := a.updateRestoreCRInVolumeStage(
+							namespacedName,
+							storkapi.ApplicationRestoreStatusInProgress,
+							storkapi.ApplicationRestoreStageVolumes,
+							msg,
+							restoreVolumeInfos,
+						)
+						if updateErr != nil {
+							logrus.Warnf("failed to update restore status: %v", updateErr)
+						}
+						return err
+					}
 					a.recorder.Event(restore,
 						v1.EventTypeWarning,
-						string(storkapi.ApplicationRestoreStatusInProgress),
-						msg)
-
-					log.ApplicationRestoreLog(restore).Errorf(msg)
-					// Update the restore status even for failed restores when storage is busy
-					_, updateErr := a.updateRestoreCRInVolumeStage(
-						namespacedName,
-						storkapi.ApplicationRestoreStatusInProgress,
-						storkapi.ApplicationRestoreStageVolumes,
-						msg,
-						restoreVolumeInfos,
-					)
-					if updateErr != nil {
-						logrus.Warnf("failed to update restore status: %v", updateErr)
-					}
+						string(storkapi.ApplicationRestoreStatusFailed),
+						message)
+					_, err = a.updateRestoreCRInVolumeStage(namespacedName, storkapi.ApplicationRestoreStatusFailed, storkapi.ApplicationRestoreStageFinal, message, nil)
 					return err
 				}
-				a.recorder.Event(restore,
-					v1.EventTypeWarning,
-					string(storkapi.ApplicationRestoreStatusFailed),
-					message)
-				_, err = a.updateRestoreCRInVolumeStage(namespacedName, storkapi.ApplicationRestoreStatusFailed, storkapi.ApplicationRestoreStageFinal, message, nil)
-				return err
+				time.Sleep(k8sutils.RestoreVolumeBatchSleepInterval)
```

Comment:
Instead of sleep should we return back from the reconcile event and instead try the next set of batch in the next reconcile event? Basically sleeping here is going to block the reconcile event processing for a long time. Also on line 690 we are updating the CR, so there will be a reconcile event queued already.